src/legacy/core_plugins/metrics/public/components/vis_editor.js

```diff
@@ -72,6 +76,7 @@ export class VisEditor extends Component {
   isValidKueryQuery = filterQuery => {
     if (filterQuery && filterQuery.language === 'kuery') {
       try {
+        const queryOptions = npSetup.uiSettings.get('query:allowLeadingWildcards');
```

Comment:
nit: to avoid of future refactoring let's: 
 - move the following code to constructor
```
   this.coreContext = {
        appName: APP_NAME,
        uiSettings: npSetup.core.uiSettings,
        store: localStorage,
      };
```
-  Now we can remove reference to npSetup
`  const queryOptions = this.coreContext.uiSettings.get('query:allowLeadingWildcards');`